azure-iot-device/tests/common/test_mqtt_transport.py

```diff
@@ -247,6 +359,19 @@ def test_event_handler_callback_raises_base_exception(
                 client=mock_mqtt_client, userdata=None, flags=None, rc=fake_rc
             )
 
+    @pytest.mark.parametrize(
+        "error_params",
+        operation_return_codes,
+        ids=["{}->{}".format(x["name"], x["error"].__name__) for x in operation_return_codes],
+    )
+    @pytest.mark.it("Raises an Exception if connect returns a failing rc code")
```

Comment:
>Exception [](start = 31, length = 9)

custom Exception. Otherwise this implies you're literally raising `Exception`.

Here and everywhere.